modules/hostap/src/supp_api.c

```diff
@@ -144,6 +144,52 @@ static void supp_shell_connect_status(struct k_work *work)
 	k_mutex_unlock(&wpa_supplicant_mutex);
 }
 
+static int wpa_supp_supported_channels(struct wpa_supplicant *wpa_s, uint8_t band, char **chan_list)
+{
+	struct hostapd_hw_modes *mode = NULL;
+	int i, j;
+	int offset, retval;
+	int size;
+	char *_chan_list;
+
+	if ((band & BIT(WIFI_FREQ_BAND_2_4_GHZ)) == BIT(WIFI_FREQ_BAND_2_4_GHZ)) {
+		mode = NULL;
+		for (j = 0; j < ((wpa_s)->hw).num_modes; j++) {
+			if (((wpa_s)->hw).modes[j].mode != HOSTAPD_MODE_IEEE80211A) {
+				mode = &((wpa_s)->hw).modes[j];
+				break;
+			}
+		}
+	} else if ((band & BIT(WIFI_FREQ_BAND_5_GHZ)) == BIT(WIFI_FREQ_BAND_5_GHZ)) {
+		mode = NULL;
+		for (j = 0; j < ((wpa_s)->hw).num_modes; j++) {
+			if (((wpa_s)->hw).modes[j].mode == HOSTAPD_MODE_IEEE80211A) {
+				mode = &((wpa_s)->hw).modes[j];
+				break;
+			}
+		}
```

Comment:
Even this can be refactored if you can get mode i.e., `HOSTAPD_MODE_*` from `WIFI_FREQ_BAND_*`.

Reply:
Made changes to identify hardware mode based on band.